protocols/interfaces.go

```diff
@@ -24,11 +24,12 @@ type LedgerRequest struct {
 
 // Equal checks for equality between the receiver and a second LedgerRequest
 func (l LedgerRequest) Equal(m LedgerRequest) bool {
-	return l.LedgerId == m.LedgerId &&
+	return l.ObjectiveId == m.ObjectiveId &&
```

Comment:
Outside the scope of this PR but it'd be nice if we have an easy way to test `Equal` functions that would have caught this.  Maybe using the [cmp](https://pkg.go.dev/github.com/google/go-cmp/cmp) package or reflection?

Reply:
Yes -- I think that's a good idea and doesn't contradict the advice to not used `go-cmp` in production code.